omnibus/files/private-chef-cookbooks/private-chef/libraries/private_chef.rb

```diff
@@ -178,15 +180,18 @@ def generate_secrets(node_name)
       me = PrivateChef["servers"][node_name]
       ha_guard = PrivateChef['topology'] == 'ha' && !me['bootstrap']
 
+      PrivateChef['postgresql']['db_superuser_password'] ||= generate_hex_if_bootstrap(50, ha_guard)
       PrivateChef['redis_lb']['password'] ||= generate_hex_if_bootstrap(50, ha_guard)
       PrivateChef['rabbitmq']['password'] ||= generate_hex_if_bootstrap(50, ha_guard)
       PrivateChef['rabbitmq']['jobs_password'] ||= generate_hex_if_bootstrap(50, ha_guard)
       PrivateChef['rabbitmq']['actions_password'] ||= generate_hex_if_bootstrap(50, ha_guard)
       PrivateChef['drbd']['shared_secret'] ||= generate_hex_if_bootstrap(30, ha_guard)
       PrivateChef['keepalived']['vrrp_instance_password'] ||= generate_hex_if_bootstrap(50, ha_guard)
+      # In a second or later run, opscode_erchef has become opscode-erchef...
```

Comment:
From what I could see, the only things affected will be those that have secrets in the secrets file.  The only others with underscores are 'oc_' and those are explicitly bypassed in the - > _ conversion. 

That said, on re-read through this code, it looks like it might work if I just ensure that I write out the secrets file key as 'opscode-erchef' and not 'opscode_erchef'.  Taking another look at it now.